frontend/src/components/login/Login.tsx

```diff
@@ -0,0 +1,235 @@
+"use client";
+import React, { useState } from "react";
+import {
+  Card,
+  Spacer,
+  Button,
+  Input,
+  Checkbox,
+  Link,
+  Divider,
+  CardHeader,
+} from "@nextui-org/react";
+import PeerPrepLogo from "../PeerPrepLogo";
+import SubmitLogo from "./SubmitLogo";
+import GithubLogo from "./GithubLogo";
+import GoogleLogo from "./GoogleLogo";
+import EyeHideLogo from "./EyeHideLogo";
+import EyeShowLogo from "./EyeShowLogo";
+
+export default function LoginComponent() {
+  // States
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [checkPassword, setCheckPassword] = useState("");
+  const [name, setName] = useState("");
+  const [isRemembered, setIsRemembered] = useState(false);
+
+  // Flags
+  const [isSubmitted, setIsSubmitted] = useState(false);
+  const [isPasswordVisible, setIsPasswordVisible] = useState(false);
+  const [isCheckPasswordVisible, setIsCheckPasswordVisible] = useState(false);
+  const [isSignUp, setIsSignUp] = useState(false);
+
+  // Toggles
+  const togglePasswordVisibility = () =>
+    setIsPasswordVisible(!isPasswordVisible);
+  const toggleCheckPasswordVisibility = () =>
+    setIsCheckPasswordVisible(!isCheckPasswordVisible);
+  const toggleSignUp = () => setIsSignUp(!isSignUp);
+
+  // throw new Error("Not implemented");
+
+  return (
+    <div className="flex items-center justify-center h-screen">
+      <Card className="items-center justify-center w-96 mx-auto pt-10 pb-10">
+        <form className="w-1/2">
+          <PeerPrepLogo />
+          <CardHeader className="lg font-bold justify-center">
+            PeerPrep
+          </CardHeader>
+          <Spacer y={3} />
+          <Input
+            isClearable
+            isRequired
+            fullWidth
+            onInput={(e) => {
+              setEmail(e.currentTarget.value);
+            }}
+            placeholder="Email"
+          />
+          <Spacer y={1} />
+          <Input
+            type={isPasswordVisible ? "text" : "password"}
+            placeholder="Password"
+            isClearable
+            isRequired
+            fullWidth
+            onInput={(e) => {
+              setPassword(e.currentTarget.value);
+            }}
+            endContent={
+              isPasswordVisible ? (
```

Comment:
Except for the different logos to render and a special horizontal padding "px-2" applied to the `EyeShow`, the two buttons seem to have the same properties. So one thing you can consider is to just put one button component here, and then move the ternary operator `isPasswordVisible` under the button component so that you only choose which logo to display and not which button component to render.